ohdear/resource_site.go

```diff
@@ -1,115 +1,96 @@
 package ohdear
 
 import (
-	"encoding/json"
 	"fmt"
-	"strconv"
+	"io/ioutil"
+	"log"
 
+	"github.com/articulate/ohdear-sdk/ohdear"
 	"github.com/hashicorp/terraform/helper/schema"
-	"github.com/smallnest/goreq"
 )
 
-type Site struct {
-	Url    string `json:"url"`
-	TeamId string `json:"team_id"`
-}
-
 func resourceOhdearSite() *schema.Resource {
 	return &schema.Resource{
-		Create: resourceSiteCreate,
-		Read:   resourceSiteRead,
-		Update: resourceSiteUpdate,
-		Delete: resourceSiteDelete,
-		Exists: resourceSiteExists,
+		Create: resourceOhdearSiteCreate,
+		Read:   resourceOhdearSiteRead,
+		Delete: resourceOhdearSiteDelete,
 		Schema: map[string]*schema.Schema{
+			"site_id": &schema.Schema{
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Primary Key of the site",
+			},
 			"url": &schema.Schema{
 				Type:        schema.TypeString,
 				Required:    true,
+				ForceNew:    true,
 				Description: "URL of the site to be checked",
 			},
 			"team_id": &schema.Schema{
-				Type:        schema.TypeString,
+				Type:        schema.TypeInt,
 				Required:    true,
+				ForceNew:    true,
 				Description: "ID of the team for this site",
 			},
 		},
 	}
 }
 
-func resourceSiteExists(d *schema.ResourceData, m interface{}) (bool, error) {
-	_, _, err := goreq.New().Get("https://ohdear.app/api/sites").
-		SetHeader("Authorization", "Bearer "+m.(Config).Token).
-		SetHeader("Accept", "application/json").
-		SetHeader("Content-Type", "application/json").
-		End()
-
-	if err == nil {
-		return true, nil
-	} else {
-		return false, nil
+func resourceOhdearSiteExists(d *schema.ResourceData, meta interface{}) (bool, error) {
```

Comment:
hmm I think I am looking at the latest diff, looks like this isn't corrected, is it? 404 scenario not attended to (Terraforrm will consider it an error since you are returning false and the error and thus will blow up). Ignore me if I am missing the latest diff!!